test/mocha-junit-reporter-spec.js

```diff
@@ -556,4 +556,103 @@ describe('mocha-junit-reporter', function() {
+      var reporter = configureReporter({jenkinsMode: true }, suites);
+      var xml = reporter.getXml(reporter.suites);
+      var xsd = fs.readFileSync(path.join(__dirname, 'resources', 'jenkins-junit.xsd'));
+
+      var xsdDoc = libxml.parseXml(xsd);
+      var xmlDoc = libxml.parseXml(xml);
+
+      xmlDoc.validate(xsdDoc);
+
+      expect(xmlDoc.validationErrors).to.be.deep.equal([]);
+    });
+
+    it('generates Jenkins compatible XML when in antMode', function() {
+      var reporter = configureReporter({antMode: true }, suites);
+      var xml = reporter.getXml(reporter.suites);
+      var xsd = fs.readFileSync(path.join(__dirname, 'resources', 'JUnit.xsd'));
+
+      var xsdDoc = libxml.parseXml(xsd);
+      var xmlDoc = libxml.parseXml(xml);
+
+      xmlDoc.validate(xsdDoc);
+
+      expect(xmlDoc.validationErrors).to.be.deep.equal([]);
+    });
+
+    describe('Jenkins format', function () {
+      var suites = [
+        {
+          testsuite: {
+            title: 'Inner Suite',
+            suites: [1],
+            tests: [1]
+          },
+          pass: [ {title: 'test', fullTitle: 'Inner Suite test'} ],
+          suites: [ {
+            testsuite: {
+              title: 'Another Suite',
+              suites: [1],
+              tests: [1]
+            },
+            fail: [ {title: 'fail test', fullTitle: 'Another Suite fail test', error: new Error('failed test')}]
+          } ]
+        },
+      ];
+
+      it('generates Jenkins compatible classnames and suite name', function() {
+        var reporter = configureReporter({jenkinsMode: true}, suites);
+
+        debug('testcase', reporter.suites[0].testsuite[1].testcase[0])
+        expect(reporter.suites[0].testsuite[0]._attr.name).to.equal(suites[0].testsuite.title);
+        expect(reporter.suites[0].testsuite[1].testcase[0]._attr.name).to.equal(suites[0].pass[0].title);
+        expect(reporter.suites[0].testsuite[1].testcase[0]._attr.classname).to.equal(suites[0].testsuite.title);
+        expect(reporter.suites[1].testsuite[0]._attr.name).to.equal(suites[0].testsuite.title + '.' + suites[0].suites[0].testsuite.title);
+        expect(reporter.suites[1].testsuite[1].testcase[0]._attr.name).to.equal(suites[0].suites[0].fail[0].title);
+        expect(reporter.suites[1].testsuite[1].testcase[0]._attr.classname).to.equal(suites[0].testsuite.title + '.' + suites[0].suites[0].testsuite.title);
+      });
+    });
+
+    function configureReporter(options, suites) {
+      var reporter = createReporter(options);
+
+      reporter.flush = function(suites) {
+        reporter.suites = suites;
+      };
+
+      (suites || []).forEach(function (suite) {
+        runner.startSuite(suite.testsuite);
+        ['pass', 'fail', 'pending'].forEach(function (key) {
+          if (suite[key]) {
+            suite[key].forEach(function (test) {
+              runner[key](new Test(test.fullTitle || test.title, test.title, 1), test.error);
```

Comment:
That's what I was missing! Do you know if mocha does this too?

Reply:
Yes the parent property should be available on the real test object from mocha.
I was trying to mock mocha API enough to run the test.
But it might be better to actually run mocha progrematically and validate the results.
So if mocha changes the test object this test will fail. 
But I'm not sure how complex it will be to add this to the tests

Reply:
Good idea, tracked in #83